lib/facter/framework/detector/os_detector.rb

```diff
@@ -64,7 +64,14 @@ def detect_family
     Facter::Resolvers::OsRelease.resolve(:id_like)
   end
 
+  def detect_based_on_release_file
+    @identifier = :devuan if File.readable?('/etc/devuan_version')
+  end
+
   def detect_distro
+    detect_based_on_release_file
```

Comment:
I think this os is not as popular as the other ones, so I think that we should make this search probably before or after the SuseRelease. I noticed that we also extract the version, which is not used anywhere in the code (I could be wrong though).